src/databricks/labs/ucx/installer/policy.py

```diff
@@ -107,6 +118,42 @@ def _get_cluster_policies_with_external_hive_metastores(self):
                     yield policy
                     break
 
+    @staticmethod
+    def _extract_external_hive_metastore_sql_conf(sql_config: GetWorkspaceWarehouseConfigResponse):
+        spark_conf_dict: dict[str, str] = {}
+        instance_profile = None
+        if sql_config.instance_profile_arn is not None:
+            instance_profile = sql_config.instance_profile_arn
+            logger.info(f"Instance Profile is Set to {instance_profile}")
+        if sql_config.data_access_config is None:
+            return instance_profile, spark_conf_dict
+        for conf in sql_config.data_access_config:
+            if conf.key is None:
+                continue
+            if conf.value is None:
+                continue
+            if (
+                conf.key.startswith("spark.sql.hive.metastore")
+                or conf.key.startswith("spark.hadoop.javax.jdo.option")
+                or conf.key.startswith("spark.databricks.hive.metastore")
+                or conf.key.startswith("spark.hadoop.hive.metastore.glue")
+            ):
+                spark_conf_dict[conf.key] = conf.value
+        return instance_profile, spark_conf_dict
+
+    def _get_warehouse_config_with_external_hive_metastore(self) -> GetWorkspaceWarehouseConfigResponse | None:
+        sql_config = self._ws.warehouses.get_workspace_warehouse_config()
+        if sql_config.data_access_config is None:
+            return None
+        for conf in sql_config.data_access_config:
+            if conf.key is None:
+                continue
+            if conf.key.startswith("spark.sql.hive.metastore") or conf.key.startswith(
+                "spark.databricks.hive.metastore.glueCatalog.enabled"
+            ):
```

Comment:
```suggestion
            is_glue = conf.key.startswith("spark.databricks.hive.metastore.glueCatalog.enabled")
            if conf.key.startswith("spark.sql.hive.metastore") or is_glue:
```
for some reason i don't like the single-argument calls on multiple lines